VultisigApp/VultisigApp/Services/BlockChainService.swift

```diff
@@ -72,21 +77,43 @@ final class BlockChainService {
     }
     @MainActor
     func fetchSpecific(tx: SwapTransaction) async throws -> BlockChainSpecific {
-        let cacheKey =  getCacheKey(for: tx.fromCoin,
-                                          action: .swap,
-                                          sendMaxAmount: false,
-                                          isDeposit: tx.isDeposit,
-                                          transactionType: .unspecified,
-                                          fromAddress: tx.fromCoin.address,
-                                          feeMode: .fast)
-        if let localCacheItem =  self.localCache.get(cacheKey) {
+        
+        let cacheKey = getCacheKey(for: tx.fromCoin,
+                                    action: .swap,
+                                    sendMaxAmount: false,
+                                    isDeposit: tx.isDeposit,
+                                    transactionType: .unspecified,
+                                    fromAddress: tx.fromCoin.address,
+                                    feeMode: .fast)
+        if let localCacheItem = self.localCache.get(cacheKey) {
             let cacheSeconds = getCacheSeconds(chain: tx.fromCoin.chain)
             // use the cache item
             if localCacheItem.date.addingTimeInterval(cacheSeconds) > Date() {
                 return localCacheItem.blockSpecific
             }
         }
         
+        let fromCoin = await tx.fromCoin
+        let toCoin = await tx.toCoin
+        
+        if (fromCoin.chain == .thorChain && toCoin.chain == .base) {
+            
+            let specific = try await fetchSpecific(
```

Comment:
also , why branch out here? the only difference is tx.isDeposit 
then fix it appropriately